stats/cloud/client.go

```diff
@@ -107,6 +107,8 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
 		}
 	}
 
+	c.prepareHeader(req)
```

Comment:
Shouldn't this be called in `NewRequest()`?

Reply:
And also, the idempotency key can be set in it as well, right?

Reply:
> And also, the idempotency key can be set in it as well, right?

Yes, that's what I think in original, but we decide to add idempotency key in NewRequest, so not sure for now.

Reply:
Well, is there any reason we can't move this `c.prepareHeader(req)` call in `NewRequest()`, and move the idempotency key setting in `prepareHeader()`?

Reply:
Good point.

I still prefer doing prepare header in do, as this is Client own headers, it's better to not expose to the caller.

Reply:
>  as this is Client own headers, it's better to not expose to the caller.

Sorry, I don't understand what you mean, can you elaborate?

Reply:
> > as this is Client own headers, it's better to not expose to the caller.
> 
> Sorry, I don't understand what you mean, can you elaborate?

I mean we can moving idempotency key setting to prepareHeaders, then only call prepareHeaders before retries loop.

Never mind, let call prepareHeaders in NewRequest for now.